runtime/src/main/java/org/corfudb/protocols/logprotocol/MultiObjectSMREntry.java

```diff
@@ -92,8 +91,11 @@ public void deserializeBuffer(ByteBuf b, CorfuRuntime rt) {
         for (int i = 0; i < numStreams; i++) {
             UUID streamId = new UUID(b.readLong(), b.readLong());
 
-            int updateLength = b.readInt();
-            byte[] streamUpdates = new byte[updateLength];
+            int start = b.readerIndex();
+            MultiSMREntry.seekToEnd(b);
+            int multiSMRLen = b.readerIndex() - start;
+            b.readerIndex(start);
+            byte[] streamUpdates = new byte[multiSMRLen];
```

Comment:
do we really need a copy constructor here?

Reply:
I can slice the ByteBuf and avoid this allocation/copy, but that would  hold a reference to the parent buffer. That the parent buffer, which can be much larger, can't be de-allocated and reused.